mcux/CMakeLists.txt

```diff
@@ -43,10 +43,12 @@ endif()
 # fsl_device_registers.h and ${MCUX_DEVICE}_features.h
 zephyr_compile_definitions(${MCUX_CPU})
 
+zephyr_compile_definitions_ifdef(CONFIG_CLOCK_CONTROL_IMX_SCMI FSL_SDK_DISABLE_DRIVER_CLOCK_CONTROL=1)
```

Comment:
reuse CONFIG_HAS_IMX_SM?
zephyr_compile_definitions_ifdef(CONFIG_HAS_IMX_SM FSL_SDK_DISABLE_DRIVER_CLOCK_CONTROL=1)

Reply:
I prefer the CONFIG_CLOCK_CONTROL_IMX_SCMI, currently the implementation leverages the imx-sm SCMI transport/protocol layers, but finally, as plan, we will implement SCMI framwork on zephyr and drop the imx-sm code.